### PR TITLE
feat#10614: consolidate 'OpenMetadata' service settings under one section.

### LINKED ISSUE #10614: Consolidate Metadata -> OpenMetadata settings under one section
**Feature**
OpenMetadata Service comes under Metadata -> Services. 
OpenMetadata Service is a system-level service; we do not want to show this under Metadata Services.

1. Create a section of OpenMetadata
2. under OpenMetadata we need following sections 
     1. Search
     2. DataInsights
     3. Email
 3. DataInsights should've a pipeline deployment. We need to document in main DataInsights page if we do not have a pipeline deployed. We should show relevant documents and ask users to deploy by going to these settings page
 4. Search, similar to what we have currently, where users can trigger on-demand but we also need allow users to deploy the Pipeline to periodically schedule the indexing
 5. Email, all of email settings in config right now. It is hard to configure that way we should move all of the settings to the UI.

**Describe the task**

### PR DESCRIPTION
**Pull request by aniketkatkar97**
### Describe your changes:

Fixes #10614 

I worked on adding "Search", "Data Insight" and "Email" settings options under the "OpenMetadata" section.

**Search:**

https://user-images.githubusercontent.com/51777795/231392166-9aff87af-d032-4fbd-8f8d-6576f49381ba.mov

<hr/>

**Data Insight:**

https://user-images.githubusercontent.com/51777795/230929938-adb610fe-7c78-4624-a453-1f257feeb036.mov

<hr/>

**Email:**

https://user-images.githubusercontent.com/51777795/230864414-927c4d56-20bd-4a68-8842-e185e62a40e2.mov

<hr/>

#
### Type of change:

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [x] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx [hunk 1]
```diff
@@ -621,7 +639,7 @@ const AddIngestion = ({
 
     return (
       <span>
-        <span className="tw-mr-1 tw-font-semibold">
+        <span className="tw-mr-1 tw-font-semibold overflow-wrap-anywhere">
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx:642]: Use AntD Typography component and it's ellipsis props

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/AddIngestionButton.component.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  getIngestionTypes,
+  getSupportedPipelineTypes,
+} from 'utils/IngestionUtils';
+import { PIPELINE_TYPE_LOCALIZATION } from '../../constants/Ingestions.constant';
+import { MetadataServiceType } from '../../generated/api/services/createMetadataService';
+import { PipelineType } from '../../generated/entity/services/ingestionPipelines/ingestionPipeline';
+import { getAddIngestionPath } from '../../utils/RouterUtils';
+import { dropdownIcon as DropdownIcon } from '../../utils/svgconstant';
+import DropDownList from '../dropdown/DropDownList';
+import { AddIngestionButtonProps } from './ingestion.interface';
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/AddIngestionButton.component.tsx:30]:
```suggestion
import { AddIngestionButtonProps } from './AddIngestionButton.interface';
```

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/AddIngestionButton.component.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+              }}
+            />
+          ) : (
+            <DropdownIcon
+              style={{
+                color: '#fff',
+                verticalAlign: 'middle',
+              }}
+            />
+          )}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/AddIngestionButton.component.tsx:122]: Use button's icon prop instead

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/IngestionListTable.component.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  onIngestionWorkflowsUpdate,
+  servicePermission,
+  serviceCategory,
+  serviceName,
+  handleDeleteSelection,
+  handleIsConfirmationModalOpen,
+  ingestionData,
+  deleteSelection,
+  permissions,
+  pipelineNameColWidth,
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/IngestionListTable.component.tsx:48]: I feel we can reduce this props from parent, Can you check here

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/IngestionListTable.component.tsx [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+                disabled={!(permissions.ViewAll || permissions.ViewBasic)}
+                href={`${airflowEndpoint}/tree?dag_id=${text}`}
+                rel="noopener noreferrer"
+                target="_blank">
+                {getEntityName(record)}
+              </Typography.Link>
+            </Tooltip>
+          ) : (
+            getEntityName(record)
+          ),
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/IngestionListTable.component.tsx:81]: Take render outside of useMemo

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/IngestionListTable.component.tsx [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+                  })}
+                </div>
+              }
+              placement="bottom"
+              trigger="hover">
+              <span>{record.airflowConfig.scheduleInterval}</span>
+            </Popover>
+          ) : (
+            <span>--</span>
+          ),
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/IngestionListTable.component.tsx:109]: Create renderer function above sepratly

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/IngestionListTable.component.tsx [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+            handleIsConfirmationModalOpen={handleIsConfirmationModalOpen}
+            isRequiredDetailsAvailable={isRequiredDetailsAvailable}
+            record={record}
+            serviceCategory={serviceCategory}
+            serviceName={serviceName}
+            servicePermission={servicePermission}
+            triggerIngestion={triggerIngestion}
+            onIngestionWorkflowsUpdate={onIngestionWorkflowsUpdate}
+          />
+        ),
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/IngestionListTable.component.tsx:139]: Same

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/PipelineActions.component.tsx [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+  const handleTriggerIngestion = (id: string, displayName: string) => {
+    setCurrTriggerId({ id, state: 'waiting' });
+    triggerIngestion(id, displayName)
+      .then(() => {
+        setCurrTriggerId({ id, state: 'success' });
+        setTimeout(() => {
+          setCurrTriggerId({ id: '', state: '' });
+          showSuccessToast(t('message.pipeline-trigger-success-message'));
+        }, 1500);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/PipelineActions.component.tsx:59]: Why we have settimeout here for success message?

File: openmetadata-ui/src/main/resources/ui/src/components/SettingsIngestion/SettingsIngestion.component.tsx [hunk 9]
```diff
@@ -0,0 +1,308 @@
...
+  const [ingestionCurrentPage, setIngestionCurrentPage] = useState(1);
+
+  const serviceCategory = ServiceCategory.METADATA_SERVICES;
+  const serviceFQN = OPENMETADATA;
+
+  const getAirflowEndpoint = async () => {
+    try {
+      setIsLoading(true);
+      const res = await fetchAirflowConfig();
+      setAirflowEndpoint(res.apiEndpoint);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/SettingsIngestion/SettingsIngestion.component.tsx:68]: I feel we have airflow hook we can use that https://github.com/open-metadata/OpenMetadata/blob/cfacbfa16813b614bd2f9dd218d03caf60e7a539/openmetadata-ui/src/main/resources/ui/src/hooks/useAirflowStatus.ts

File: openmetadata-ui/src/main/resources/ui/src/pages/EditConnectionFormPage/EditConnectionFormPage.component.tsx [hunk 10]
```diff
@@ -102,47 +108,53 @@ function EditConnectionFormPage() {
...
+            setServiceDetails(resService);
+            setSlashedBreadcrumb([
+              {
+                name: startCase(serviceCategory),
+                url: getSettingPath(
+                  GlobalSettingsMenuCategory.SERVICES,
+                  getServiceRouteFromServiceType(
+                    serviceCategory as ServiceTypes
+                  )
+                ),
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/EditConnectionFormPage/EditConnectionFormPage.component.tsx:125]: Always call separate function from useEffect to make useEffects more readble

File: openmetadata-ui/src/main/resources/ui/src/components/SettingsIngestion/SettingsIngestion.component.tsx (hunk 9 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/SettingsIngestion/SettingsIngestion.component.tsx:68]: will take a look

Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/SettingsIngestion/SettingsIngestion.component.tsx:68]: @chirag-madlani the hook that we have is to check the airflow status with a different API call. Here we are fetching the airflowEndpoint

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/AddIngestionButton.component.tsx (hunk 3 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/Ingestion/AddIngestionButton.component.tsx:122]: Didn't use the icon prop of the button as it doesn't allow configuring the position of the icon to display.

**Comment by harshach** [comment]
@aniketkatkar97 rename "Add Elastic Search Reindex Ingestion" -> "Deploy Search Index Pipeline" 
There should be three tabs "On Demand" , "Live", "Scheduled" and move the stream indexing status to Live tab

**Comment by harshach** [comment]
Instead of showing "No Index Data available" when there is no pipeline deployed. Let's add text around what is Search Index Pipeline is and What is Data Insights Pipeline. Link them to appropriate docs

**Comment by harshach** [comment]
For Email, instead of opening up a dialog box . Let's do what we are doing in ingestion add a form and detailed setup instructions right next to it. cc @Sachin-chaurasiya

File: openmetadata-ui/src/main/resources/ui/src/components/TriggerReIndexing/TriggerReIndexing.component.tsx [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+  EventPublisherJob,
+  RunMode,
+} from '../../generated/system/eventPublisherJob';
+import { getJobDetailsCard } from '../../utils/EventPublisherUtils';
+import { showErrorToast, showSuccessToast } from '../../utils/ToastUtils';
+import ReIndexAllModal from './ElasticSearchReIndexModal.component';
+
+function TriggerReIndexing() {
+  const { t } = useTranslation();
+  const { fqn } = useParams<{ fqn: string }>();
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TriggerReIndexing/TriggerReIndexing.component.tsx:43]:
```suggestion
  const { tab } = useParams<{ tab: string }>();
```

File: openmetadata-ui/src/main/resources/ui/src/components/TriggerReIndexing/TriggerReIndexing.component.tsx [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+  const { socket } = useWebSocketConnector();
+
+  const isOnDemandTab = useMemo(
+    () => fqn === ELASTIC_SEARCH_RE_INDEX_PAGE_TABS.ON_DEMAND,
+    [fqn]
+  );
+
+  const handleModalVisibility = (state: boolean) => {
+    setIsModalOpen(state);
+  };
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TriggerReIndexing/TriggerReIndexing.component.tsx:62]: We can remove this function and direct use state handler

File: openmetadata-ui/src/main/resources/ui/src/constants/EmailConfig.constants.ts [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { TransportationStrategy } from 'generated/email/smtpSettings';
+
+export const TRANSPORTATION_STRATEGY_OPTIONS = [
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/constants/EmailConfig.constants.ts:16]: We can directly use `TransportationStrategy` if that will be the source of truth
Get rid of overhead types

File: openmetadata-ui/src/main/resources/ui/src/pages/EditEmailConfigPage/EditEmailConfigPage.component.tsx [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+    );
+  }, [history]);
+
+  useEffect(() => {
+    fetchEmailConfigValues();
+  }, []);
+
+  return (
+    <PageContainerV1>
+      <div className="self-center w-min-60">
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/EditEmailConfigPage/EditEmailConfigPage.component.tsx:150]: Why we need this? PageLayoutV1 should provide you all the layouts needed

**Review by chirag-madlani — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx [hunk 15]
```diff
@@ -126,37 +126,51 @@ export type ServicePageData =
   | Container
   | DashboardDataModel;
 
+const tableComponent = {
+  body: {
+    row: ({ children }: { children: React.ReactNode }) => (
+      <tr data-testid="row">{children}</tr>
+    ),
+  },
+};
```
[the remark this answers is not in the document]
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx:135]: In these changes, I just brought it outside and stored it in `tableComponent` variable. I believe those changes were done for the ability to give `data-testid` to the `tr` tag

File: openmetadata-ui/src/main/resources/ui/src/components/TriggerReIndexing/TriggerReIndexing.component.tsx (hunk 11 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/TriggerReIndexing/TriggerReIndexing.component.tsx:43]: We are already using the tab parameter in the URL to indicate the menu item selected in the settings left panel, and also the path with the fqn parameter is already in use in Teams page

File: openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx (hunk 15 above)
[the remark this answers is not in the document]
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/pages/service/index.tsx:135]: We can get rid of that while refactoring afterward

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**